server/src/main/java/org/opensearch/cluster/service/TaskBatcher.java

```diff
@@ -90,20 +91,33 @@ public void submitTasks(List<? extends BatchedTask> tasks, @Nullable TimeValue t
                     throw new IllegalStateException("cannot add duplicate task: " + a);
                 }, IdentityHashMap::new));
```

Comment:
It used to be IdentityHashMap which would uses reference equality instead == or equals but now with LinkedHashSet, it would change the logic. Did you test by putting two tasks whose equals would return true but reference equality would return false.

Reply:
Yes @shwetathareja , I have tested it. Updated UT as well to test this scenario.

Reply:
> two tasks whose equals would return true and their reference equality would return false as those are 2 different objects [this should pass]

@dhwanilpatel it is working with ```SimpleTask``` class instantiated in the test code as ```hashCode()``` method is returning different value for different objects even when their equals method return true. Please override hashCode and then the above wouldn't work

Reply:
Thanks @shwetathareja  for catching it, I have missed that. It was failing after changing `hashCode()` as well.

I have removed use of `LinkedHashSet` and used `IdentityHashMap` itself to keep the logic same.